libcxx/include/__config

```diff
@@ -1427,20 +1427,6 @@ __sanitizer_verify_double_ended_contiguous_container(const void*, const void*, c
 #    define _LIBCPP_NODISCARD
 #  endif
 
-// _LIBCPP_NODISCARD_EXT may be used to apply [[nodiscard]] to entities not
-// specified as such as an extension.
-#  if !defined(_LIBCPP_DISABLE_NODISCARD_EXT)
-#    define _LIBCPP_NODISCARD_EXT _LIBCPP_NODISCARD
-#  else
-#    define _LIBCPP_NODISCARD_EXT
-#  endif
-
-#  if _LIBCPP_STD_VER >= 20 || !defined(_LIBCPP_DISABLE_NODISCARD_EXT)
-#    define _LIBCPP_NODISCARD_AFTER_CXX17 _LIBCPP_NODISCARD
-#  else
-#    define _LIBCPP_NODISCARD_AFTER_CXX17
-#  endif
```

Comment:
Since this might break patches in flight and downstream vendors without a transition I feel we should give this change more visibility before landing.

Alternatively; how do you feel about about making these macros deprecated for 1 release cycle?

Reply:
CC @llvm/libcxx-vendors

If nobody complains I think we can just remove them.

I don't see anybody using `_LIBCPP_NODISCARD_AFTER_CXX17` downstream, since it's been there for applications in the standard. `_LIBCPP_NODISCARD_EXT` might be used, but the fix is to replace `_LIBCPP_NODISCARD_EXT` with `_LIBCPP_NODISCARD`, which is a task of a few minutes.

Reply:
Thanks for the info. Then let's wait and see whether vendors object. If not I'm happy to keep this as is. I leave the final approval to @ldionne. Again I'm happy with the patch and this approach.